### PR TITLE
Break OggDemuxer::readPage loop on EOF.

### LINKED ISSUE #1728: Löve crashes/hangs if trying to make a video from a non video file
To reproduce, just create a random file "a" and try to run the code:
```lua
love.graphics.newVideo("a")
```

### PR DESCRIPTION
**Pull request by MikuAuahDark**
This fixes #1728

### CODE REVIEW

**Comment by slime73** [comment]
The theora decoding code is really hard to follow.. I'll need to test this on a few videos I think (especially ones that have caused problems in the past) before knowing if it's safe to merge

**Comment by rameshvarun** [comment]
This is related to a long standing issue that I initially reported in #1406 and still occurs on master. My understanding:

- In some edge cases we fail to establish sync before the EOF.
- readPage hangs in an infinite loop with file->read always returning 0

This PR doesn't fully fix that problem but I think it's in the right direction. Right now it solves the case where we can't establish sync during initialization.

File: src/modules/video/theora/OggDemuxer.cpp [hunk 1]
```diff
@@ -118,22 +121,23 @@ OggDemuxer::StreamType OggDemuxer::findStream()
...
 		ogg_stream_clear(&stream);
 		ogg_sync_reset(&sync);
 	}
 
-	streamInited = true;
 	while (true)
 	{
 		// If this page isn't at the start of a stream, we've seen all streams
-		readPage();
+		readPage(true);
```
Comment (slime73) [inline src/modules/video/theora/OggDemuxer.cpp:133]: Maybe it'd be better for readPage to return something instead of optionally throwing an exception? And then this code could `return STREAM_UNKNOWN` if readPage returns a bad value, or something.

I only suggest it because `TheoraVideoStream`'s constructor which calls `findStream` checks its return value and throws a slightly more detailed exception, already.
